src/recon.rs

```diff
@@ -3906,7 +3942,11 @@ pub(crate) unsafe fn rav1d_recon_b_inter<BD: BitDepth>(
                             (*ts).tiling.col_end >> ss_hor,
                             (*ts).tiling.row_end >> ss_ver,
                             0 as EdgeFlags,
-                            uvdst,
+                            slice::from_raw_parts(
+                                ((*f).cur.data[1 + pl as usize] as *const BD::Pixel)
+                                    .offset(cmp::min(data_diff, 0)),
+                                data_diff.abs() as usize + data_width as usize,
+                            ),
```

Comment:
To keep the "distance" from the existing code smaller, could you put this in a `uvdst_slice` variable or something, and then pass `uvdst_slice` as the argument?  And the same for the other similar cases in this commit.  I think it will help when moving the other uses of `uvdst`, etc. to the safe, slice version.

Overall, though, this is a really great change, and is very helpful in understanding the layout and index/length calculations on the pixel `data` better.  And it's exactly the right way to make `fn rav1d_prepare_intra_edges` fully safe.  Thanks!